core/v2/types.go

```diff
@@ -0,0 +1,99 @@
+package corev2
+
+import (
+	"math"
+
+	"github.com/Layr-Labs/eigenda/encoding"
+)
+
+var (
+	ParametersMap = map[uint8]BlobVersionParameters{
+		0: {CodingRate: 8, ReconstructionThreshold: 0.22, NumChunks: 8192},
+	}
+)
+
+// Assignment contains information about the set of chunks that a specific node will receive
+type Assignment struct {
+	StartIndex uint32
+	NumChunks  uint32
+}
+
+// GetIndices generates the list of ChunkIndices associated with a given assignment
+func (c *Assignment) GetIndices() []uint32 {
+	indices := make([]uint32, c.NumChunks)
+	for ind := range indices {
+		indices[ind] = c.StartIndex + uint32(ind)
+	}
+	return indices
+}
+
+// BlobHeader contains all metadata related to a blob including commitments and parameters for encoding
+type BlobHeader struct {
+	Version uint8
+
+	encoding.BlobCommitments
+
+	// QuorumInfos contains the quorum specific parameters for the blob
+	QuorumNumbers []uint8
+
+	// PaymentHeader contains the payment information for the blob
+	PaymentHeader
+
+	// AuthenticationData is the signature of the blob header by the account ID
+	AuthenticationData []byte `json:"authentication_data"`
+}
+
+func (b *BlobHeader) GetEncodingParams() (encoding.EncodingParams, error) {
+
+	params := ParametersMap[b.Version]
+
+	length, err := GetChunkLength(b.Version, uint32(b.Length))
+	if err != nil {
+		return encoding.EncodingParams{}, err
+	}
+
+	return encoding.EncodingParams{
+		NumChunks:   uint64(params.NumChunks),
+		ChunkLength: uint64(length),
+	}, nil
+
+}
+
+type PaymentHeader struct {
+	// AccountID is the account that is paying for the blob to be stored. AccountID is hexadecimal representation of the ECDSA public key
+	AccountID string
+
+	// Cumulative Payment
+	CumulativePayment uint64
+
+	BinIndex uint64
+}
+
+type BlobCertificate struct {
+	BlobHeader
+
+	// ReferenceBlockNumber is the block number of the block at which the operator state will be referenced
+	ReferenceBlockNumber uint64
+
+	// RelayKeys
+	RelayKeys []uint16
+}
+
+type BlobVersionParameters struct {
+	CodingRate              uint32
+	ReconstructionThreshold float64
+	NumChunks               uint32
+}
+
+func (p BlobVersionParameters) MaxNumOperators() uint32 {
```

Comment:
This is theoretical max number of operators based on the config, which is different from the max number of operators imposed on contracts. Do we need this method or just use what's on the contracts? 
It's problematic if a `BlobVersionParameters` is set in a way that allows max # operators smaller than what's configured onchain, so it looks like we already have to make sure the set of parameters allow the max # operators configured onchain when they're added

Reply:
Good callout. Since all of this will be on chain, what about having some on-chain checks to ensure consistency when any of these parameters are updated?

Reply:
That's a good idea. Can you add a quick comment over `BlobVersionParameters` that this is supposed to go onchain?

cc @0x0aa0

Reply:
Done